### ISSUE TITLE
Year 1 and Year 2 versions of the same weapon aren't considered duplicates

### ISSUE THREAD

**Issue by kshep** [DestinyItemManager/DIM#3027]
For example, I have a year one Antiope and 4 year two Antiopes. When I search for "is:dupe" the 4 year two weapons are highlighted but the year one is not. It seems like the year one weapon could be considered a dupe that simply has a smaller set of perks.

**Comment by kyleshay** [comment]
Seeing the same. It is because the weapons themselves have different hash ids (which is what we use to group items for duplicates.)

Ballyhoo Mk.27
[Launch #653875712](https://db.destinytracker.com/d2/en/items/653875712)
[Forsaken #2591586263](https://db.destinytracker.com/d2/en/items/2591586263)

**Comment by kyleshay** [comment]
Technically... it does work if we group by name, but that is likely to be brittle. 🤔

Think we'd need a second opinion before we made that change. If the names are the same, then yeah... they're the same items. 🤷‍♂️

<img width="637" alt="screen shot 2018-09-10 at 8 41 12 pm" src="https://user-images.githubusercontent.com/424158/45336798-e1271b80-b539-11e8-92a6-ec3f365d5755.png">

**Comment by philkernick** [comment]
Except they are not.  One of the key things about "sameness" has a direct in-game effect on infusion.  If you infuse an item into the same item, it costs only glimmer.  If you infuse two different items, it costs glimmer, materials and your first born child!

I've tested in-game, and a Y1 Better Devils is not considered the same as a Y2 Better Devils for infusion purposes.  If we report them as same, I expect someone to raise an issue saying that they really aren't.

So I think we should retain the current behaviour.

**Comment by kshep** [comment]
Leave it to Bungie to put us in a position where the meaning of "duplicate" is a matter of opinion :)

Using "is:dupe" to find infusion fuel feels strange to me and is definitely something new for Forsaken, right? I still remember it primarily as a feature to let you compare perks on random rolls back in D1, and as my primary tool for clearing out inventory in D2 or to make sure I had 'one of each' element for Energy and Power weapons, i.e. compare perks.

Considering all Better Devils to be duplicates definitely seems more intuitive to me, but maybe we could add something like "is:dupename" vs "is:dupehash"?

**Comment by duckmanBR** [comment]
Have no opinion about this one, but just don't forget that if we change this, we must be sure that the compare window will also reflect this change.

**Comment by bhollis** [comment]
If they can't infuse together as duplicates I think I vote against including them as dupes.

**Comment by kyleshay** [comment]
Agreed Ben, suppose we could add an additional filter but by name... But honestly, I think I'm (and others, probably ) are going to scrap most non-forsaken gear. We should re open and revisit this in a month or two to see if people still are holding old gear

**Comment by kyleshay** [comment]
Thanks kshep for the suggestion and thank you Phil for infusing to test!

**Comment by kshep** [comment]
> We should re open and revisit this in a month or two to see if people still are holding old gear

I'll probably take this as a sign I should just shard all the old gear. Too much of a headache to try to track stuff if it's not supported in DIM

**Comment by Wh1rledPeas** [comment]
<img src="https://user-images.githubusercontent.com/3685437/49315969-65e06080-f4bd-11e8-8ad7-48b295024a91.png" width="200" align="right" />

> I've tested in-game, and a Y1 Better Devils is not considered the same as a Y2 Better Devils for infusion purposes. 

I'm not sure what you're referring to, but I have been able to infuse Y2 Better Devils into Y1, for 5k Glimmer only. Perhaps this behavior is different now?

I found this ticket because I was coming to report the same thing. I would like to see Y1 items that are named the same, show up in the "duplicate grouping' of the infusion tool. There are some Y1 weapons that I have masterworks of, that are more valuable to me as a masterwork, than with new random rolls.
